### PR TITLE
fix: `train_on_inputs: true` ignored for sharegpt

### PR DESCRIPTION
**Pull request by NanoCode012**
Following this discussion here: https://discord.com/channels/1104757954588196865/1192621815172964522

Sharegpt conversation masks all the time and does not respect `train_on_inputs`. This PR fixes it.

### CODE REVIEW

File: src/axolotl/prompt_tokenizers.py [hunk 1]
```diff
@@ -402,18 +404,24 @@ def tokenize_prompt(self, prompt):
...
-                    # not masked out from labels
                     labels = copy.deepcopy(res["input_ids"])
-                    len_role = len(role_res["input_ids"])
-                    labels[:len_role] = [IGNORE_TOKEN_ID] * min(len_role, len(labels))
+                    if not self.train_on_inputs:
+                        # mask out role tokens from the labels
+                        len_role = len(role_res["input_ids"])
+                        labels[:len_role] = [IGNORE_TOKEN_ID] * min(
+                            len_role, len(labels)
+                        )
```
Comment (NanoCode012) [inline src/axolotl/prompt_tokenizers.py:417]:
```suggestion
                        len_role = len(role_res["input_ids"])
                        labels[:len_role] = [IGNORE_TOKEN_ID] * len_role
```

Can I do this? I'm not sure why there was a `min`.

**Comment by winglian** [comment]
It would be good to have unit tests that verify this functionality. I can help with this over the weekend.

File: src/axolotl/prompt_tokenizers.py (hunk 1 above)
Comment (winglian) [reply src/axolotl/prompt_tokenizers.py:417]: this was introduced in https://github.com/OpenAccess-AI-Collective/axolotl/pull/723, I'd prefer to leave it for now and prevent a regression.

**Comment by winglian** [comment]
@NanoCode012 I added a test for this to #1054 , but it's commented out for now since it's failing until this is rebased over that

**Comment by NanoCode012** [comment]
You can merge that first then. This can come after

**Comment by NanoCode012** [comment]
Looks good!

**Pull request merged**